### PR TITLE
Develop vertical tweak

### PR DESCRIPTION
**Pull request by osamuaoki**
I ported patches from main to devel.

I was surprised normal <C-L> didn't work.  I guess there must be some config parameter somewhere since I sill see the configuration for window jumping.  :-)

### CODE REVIEW

File: lua/doom/core/functions/init.lua [hunk 1]
```diff
@@ -494,17 +494,83 @@ M.edit_config = function()
...
   elseif selected_config ~= 0 then
     log.error("Invalid option selected.")
   end
 end
 
+-- set_indent set the indent and tab related numbers
+M.set_indent = function()
+  local indent = tonumber(vim.fn.input("Set all tab related options to a specified number and set expandtab\n(0 to reset to vim defaults, ? to print current settings): "))
+  if (indent == nil) or (indent < 0) then
+    vim.cmd("set expandtab? softtabstop? tabstop? shiftwidth? expandtab?")
```
Comment (NTBBloodbath) [inline lua/doom/core/functions/init.lua:515]: Here's a duplicated `expandtab?`

**Review by NTBBloodbath — CHANGES_REQUESTED** [review]
Thank you so much per doing this, I'm really loving all the changes!

Just a small thing to be fixed and we will be ready to go :)

**Comment by osamuaoki** [comment]
Except for completion, here is my tweak patch and vertical split patch organized in 2 separate series commits.

**Comment by osamuaoki** [comment]
Oh, your requested change was done.  I did more.  Good night.

**Review by NTBBloodbath — APPROVED** [review]

**Comment by NTBBloodbath** [comment]
Thank you so much, should I merge this now and add the completion patch later in another PR?

**Comment by osamuaoki** [comment]
I now see completion may be toggled.  Let me experiment.

**Comment by osamuaoki** [comment]
Please wait.  Now I know you like these, I realize I need to refine code for multiple use cases.  (For example, depending on package install state, change available tweak option... etc.)

By the way, you make simple merges for these topic branches.  If you use rebase when you pull these topic branches, then logical history is easier to track, I think. 

Also question for topic branch management.  I don't know if you like seeing force updates, squash history, and always rebased branch from the latest develop branch?

In the mean time, I will separate gitsigns patch from this topic branch. That can be merged immediately.

**Comment by NTBBloodbath** [comment]
> Please wait.  Now I know you like these, I realize I need to refine code for multiple use cases.  (For example, depending on package install state, change available tweak option... etc.)

Sure thing, there's no hurry. It's my time to sleep so take your time haha.

> By the way, you make simple merges for these topic branches.  If you use rebase when you pull these topic branches, then logical history is easier to track, I think. 

Hmm yeah I think you're right. Sometimes I merge or squash but now that you mention this I think rebase could be a better approach for us, especially for retaining a cleaner history and having a more efficient incorporation of the new commits.

> Also question for topic branch management.  I don't know if you like seeing force updates, squash history, and always rebased branch from the latest develop branch?

I'm ok with forced updates if they're necessary and squash history if changes aren't too big so we can have a better tracking of the major changes and rebased branches are also pretty fine.

I think we should maintain a good commits history and rebases are the best approach for this as far as I know.

> In the mean time, I will separate gitsigns patch from this topic branch. That can be merged immediately.

Alright, thank you!

**Comment by osamuaoki** [comment]
I just pushed a large updates rebased to your latest updated develop branch.

It works for me is the current state.

I think I tested for where autopairs is not installed. I didn't do the same for missing treesetter.    I think this needs to be robust.

I am going to sleep.  So please review and test but don't merge.

Thanks.

**Review by NTBBloodbath — APPROVED** [review]

**Comment by NTBBloodbath** [comment]
Hi, everything looks _and works_ great on my side.

Thanks for checking the existence of autopairs plugin while testing, as you said this mechanism needs to be robust!

Also thanks for rebasing, hope this is going to be the start of a better commits history :)

**Comment by osamuaoki** [comment]
Yes, completion toggle is here.

Instead of reading through the code, I cheated by asking the upstream developer to get toggle go with completion.

I also put extra protection for treesetter, too.  Since this is not flag selectable package, it is not conditional in whichkey configuration.  (I am not asking to make such flag in user configuration file.  This topic deserves separate discussion how far should we go to address user modification segregation.  I think current state is healthy.)

Now that I have finished all the desirable features, after cool headed review, this is ready to be merged.

Osamu

**Comment by osamuaoki** [comment]
If you wants me to rebase, let me now.  (I think it is conflict free rebase.)

**Review by NTBBloodbath — APPROVED** [review]

**Review by NTBBloodbath — APPROVED** [review]

**Comment by NTBBloodbath** [comment]
> Yes, completion toggle is here.
> 
> Instead of reading through the code, I cheated by asking the upstream developer to get toggle go with completion.

Awesome, nice idea! :+1:

> I also put extra protection for treesetter, too. Since this is not flag selectable package, it is not conditional in whichkey configuration. (I am not asking to make such flag in user configuration file. This topic deserves separate discussion how far should we go to address user modification segregation. I think current state is healthy.)

Yeah we're not able to disable treesitter anymore since a few Doom versions ago, was a "design" choice. I think current protection is enough for us and shouldn't fail.

> Now that I have finished all the desirable features, after cool headed review, this is ready to be merged.
> 
> Osamu

That's great! Thank you so much for this contribution, I really appreciate it :slightly_smiling_face:

**Comment by NTBBloodbath** [comment]
> If you wants me to rebase, let me now. (I think it is conflict free rebase.)

If this doesn't annoy you then please do it :)

**Comment by osamuaoki** [comment]
I rebased.  (It was clean automatic rebase)

Please check the last commit which touches your documentation to address things discussed on "issues".

https://github.com/NTBBloodbath/doom-nvim/issues/126

**Review by NTBBloodbath — APPROVED** [review]

**Comment by NTBBloodbath** [comment]
Thank you so much mate!

**Pull request merged**